pose/airsim/pose_handlers/pose_handlers.cpp

```diff
@@ -41,18 +41,37 @@ void PoseHandlers::set_drone_pose(
     bool *exit_flag
 ) {
     uint64_t msg_count = 0;
+    std::unordered_map<uint16_t, uint64_t> messageCount;
+
     do {
+        msg_count = 0;
+
         mutex_pose_message->lock();
-        if (pose_message->message_counter != 0) {
+        // first, have we've ever seen this drone before?
+        auto it = messageCount.find(pose_message->drone_id);
+        if (it != messageCount.end()) {
+            // yes, we have seen this drone before
+            // lets record the message count
+            msg_count = it->second;
+        }
+        else {
+            // no, we have not seen this drone before
+            // let's clear the message count
+            // and spawn a new drone
+            msg_count = 0;
             spawn_unique_drone(vehicle_interface, pose_message->drone_id, pose_message->drone);
         }
 
+        // Please recall that the message count starts at 1, and increments by 1 each time
+        // also each drone has its own message count
+        // if the message count is greater than the last message count, then we have new information
         if (pose_message->message_counter > msg_count) {
```

Comment:
I wonder if messages can be received in different order than they are sent. Also, should we have a another incremental count on reciver side too? If the largest message count for that drone and total received message count is not name, that will imply packet loss.